### PR TITLE
better uniq implementation

### PR DESCRIPTION
**Pull request by jasir**
Hello,

thank you for this fantastic library.
I would like to suggest this small change in implementation.

Best regards,

Jarda

### CODE REVIEW

**Review by richytong — CHANGES_REQUESTED** [review]
In order to truly know if your implementation is better I will require benchmarks for the old uniq implementation as well as those for your implementation. You can find precedent for benchmarks in one of the benchmark files like `x/find.benchmark.js`.

**Comment by jasir** [comment]
Hello @richytong,

I have removed ramda, is magnitude slower then the other. In benchmarks, i have used different Array length (SIZE) and variability of items `<1; SCATTER-1>`

some results (size:1000, scatter:10), new 

```
uniq - original(1000, scatter: 10): 1e+5: 1.947s
uniq - lodash(1000), scatter: 10: 1e+5: 1.767s
uniq - new(1000), scatter: 10: 1e+5: 1.728s
```

```
uniq - original(1000, scatter: 10): 1e+5: 1.947s
uniq - lodash(1000), scatter: 10: 1e+5: 1.767s
uniq - new(1000), scatter: 10: 1e+5: 1.728s
```

```
uniq - original(100, scatter: 10): 1e+6: 2.291s
uniq - lodash(100), scatter: 10: 1e+6: 664.204ms
uniq - new(100), scatter: 10: 1e+6: 1.581s
```

I have runned other combinations, new implementation is generally 20% - 100% faster then old one.
I will force push with tests.

File: x/uniq.benchmark.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+const R = require('ramda')
+const _ = require('lodash')
+const isArray = require('../_internal/isArray')
+const reduce = require('../reduce')
+
+/**
+ * @name uniq
+ *
+ * @benchmark
+ * find(gt0)(array): 1e+6: 13.627ms
```
Comment (richytong) [inline x/uniq.benchmark.js:12]: please replace these with your results

**Review by richytong — CHANGES_REQUESTED** [review]

File: x/uniq.benchmark.js (hunk 1 above)
Comment (jasir) [reply x/uniq.benchmark.js]: ok, done.

**Review by richytong — APPROVED** [review]
LGTM

File: x/uniq.js [hunk 2]
```diff
@@ -1,6 +1,4 @@
 const isArray = require('../_internal/isArray')
-const reduce = require('../reduce')
-
 /**
```
Comment (richytong) [inline x/uniq.js:2]: lint i think

**Review by richytong — CHANGES_REQUESTED** [review]

**Comment by CrossEye** [comment]
@jasir:

> I have removed ramda, is magnitude slower then the other.

It would be, as it uses its value equality operation by default.  I have no idea how `R.uniqWith(R.identity)` would fare here (probably still much slower), but it would be the closest Ramda has to offering an analog.

File: x/uniq.js (hunk 2 above)
Comment (jasir) [reply x/uniq.js:3]: force pushed, hopefully this will be ok now.

**Review by richytong — CHANGES_REQUESTED** [review]
just to confim did you run the timeInLoops one at a time? I have found the results skew if you run more than one timeInLoop at a time

**Comment by jasir** [comment]
One by one.

**Pull request merged**